### PR TITLE
RIA-3088 LegalRep submits appeal with a remission

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3088](https://tools.hmcts.net/jira/browse/RIA-3088)

### Change description ###
Legal rep submits appeal with a remission

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java [hunk 1]
```diff
@@ -61,7 +60,83 @@ public boolean canHandle(
...
+                    setFeePaymentDetails(asylumCase, appealType);
+                }
+                asylumCase.clear(RP_DC_APPEAL_HEARING_OPTION);
+                break;
+
+            case DC:
+            case RP:
+                String hearingOption = asylumCase.read(RP_DC_APPEAL_HEARING_OPTION, String.class)
+                    .orElseThrow(() -> new IllegalStateException("Appeal hearing option is not present"));
+                asylumCase.write(DECISION_HEARING_FEE_OPTION, hearingOption);
```
Comment (Lee365) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java:95]: Test if this written for all appeal_types

**Review by Lee365 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java (hunk 1 above)
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java:95]: Added the assertion for DC/RP appeal types

**Review by Lee365 — DISMISSED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**